app/src/main/java/io/gnosis/safe/helpers/AddressHelper.kt

```diff
@@ -0,0 +1,35 @@
+package io.gnosis.safe.helpers
+
+
+import android.widget.TextView
+import io.gnosis.safe.utils.shortChecksumString
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import pm.gnosis.blockies.BlockiesImageView
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import javax.inject.Inject
+
+//TODO legacy handles name too
+class AddressHelper
+@Inject constructor() {
+    fun populateAddressInfo(
+        addressView: TextView,
+        imageView: BlockiesImageView?,
+        address: Solidity.Address
+    ) {
+
+        GlobalScope.launch(Dispatchers.Default) {
```

Comment:
I think it is discouraged to use GlobalScope (need to check this again)

Reply:
It is, this was copied from Kouban. I can have a closer look.